src/ott/solvers/linear/continuous_barycenter.py

```diff
@@ -107,13 +109,63 @@ def solve_linear_ot(
     )(bar_prob.weights, barycenters_per_measure)
 
     return self.set(
+        x=x_new,
         costs=updated_costs,
         linear_convergence=linear_convergence,
-        errors=errors,
-        x=x_new
+        linear_outputs=outs,
+        errors=errors
     )
 
 
+class FreeBarycenterOutput(NamedTuple):
+  """Holds the output of a Free Wasserstein barycenter solver.
+
+  Objects of this class contain both solutions and problem definition of a
+  regularized Free Barycenter OT problem, along several methods that can be used
+  to access its content.
+
+  Args:
+    x: barycenter points.
+    a: barycenter weights.
+    costs: Holds the sequence of regularized GW costs seen through the outer
+      loop of the solver.
+    linear_convergence: Holds the sequence of bool convergence flags of the
+      inner Sinkhorn iterations.
+    linear_outputs: Holds latest output objects returned by linear solver when
+      recomputing transport between barycenter vs. each of the input measures.
+    errors: Holds sequence of vectors of errors of the Sinkhorn algorithm
+      at each iteration.
+  """
+
+  x: Optional[jnp.ndarray] = None
+  a: Optional[jnp.ndarray] = None
+  bar_prob: Optional[barycenter_problem.FreeBarycenterProblem] = None
+  costs: Optional[jnp.ndarray] = None
+  linear_convergence: Optional[jnp.ndarray] = None
+  linear_outputs: Optional[Tuple[Any]] = None
+  errors: Optional[jnp.ndarray] = None
+
+  @property
+  def all_linear_solvers_converged(self):
+    """Whether all linear convergence flags converged."""
+    return jnp.all(self.linear_convergence[self.linear_convergence != -1])
+
+  def matrix_at_index(self, measure_index: int) -> jnp.ndarray:
+    """Return the transport matrix from barycenter to measure_index measure."""
+    size_measure = self.bar_prob.num_per_measure[measure_index]
+    return self.linear_output_at_index(measure_index).matrix[:, :size_measure]
```

Comment:
This might fail when `size_measure` is traced and not static, let's use dynamic slice in dim instead?

Reply:
thanks! just changed this.